### ISSUE TITLE
training mode/config should be set via functions not class decorators

### ISSUE THREAD

**Issue by jlewi** [kubeflow/fairing#39]
Currently the class decorator determines the training mode (e.g. local vs. distributed).

```
@kubeflow.DistributedTraining(worker_count=3, ps_count=1, namespace='kubeflow')
class MyModel(object):
    def train(self):
         ....
```

The downside of this approach is that the user has to change the class annotation in order to change the mode of training. I think it would be preferable to be able to change how it runs just with parameters to the train function e.g

```
config = LocalConfig()
myModel.train(config)

config = DistributedConfig(....)
myModel.train(config)
```

**Comment by jlewi** [comment]
@mattf Any update on this?

**Comment by jlewi** [comment]
@r2d4 Any update?

**Comment by r2d4** [comment]
@jlewi PTAL at #64. If that API looks good I'll go ahead and update the documentation on that PR

**Comment by r2d4** [comment]
Fixed by #64 
/close
